src/main/java/org/apache/commons/cli/help/AbstractHelpFormatter.java

```diff
@@ -0,0 +1,328 @@
+     * @param options the Options to sort.
+     * @return a new list of options ordered by the comparator.
+     */
+    public List<Option> sortedOptions(final Options options) {
+        return sortedOptions(options == null ? null : options.getOptions());
+    }
+
+    /**
+     * Creates a new list of options ordered by the comparator.
+     * @param options the Options to sort.
+     * @return a new list of options ordered by the comparator.
+     */
+    public List<Option> sortedOptions(final Iterable<Option> options) {
+        List<Option> result = new ArrayList<>();
+        if (options != null) {
+            options.forEach(result::add);
+            result.sort(comparator);
+        }
+        return result;
+    }
+
+    /**
+     * Return the string representation of the options as used in the syntax display.
+     * @param options The options to create the string representation for.
+     * @param lookup a function to determine if the Option is part of an OptionGroup that has already been processed.
+     * @return the string representation of the options as used in the syntax display.
+     */
+    protected String asSyntaxOptions(final Iterable<Option> options,
+                                          final Function<Option, OptionGroup> lookup) {
+        // list of groups that have been processed.
+        final Collection<OptionGroup> processedGroups = new ArrayList<>();
+        final List<Option> optList = sortedOptions(options);
+        StringBuilder buff = new StringBuilder();
+        String pfx = "";
+        // iterate over the options
+        for (final Option option : optList) {
+            // get the next Option
+            // check if the option is part of an OptionGroup
+            final OptionGroup group = lookup.apply(option);
+            // if the option is part of a group
+            if (group != null) {
+                // and if the group has not already been processed
+                if (!processedGroups.contains(group)) {
+                    // add the group to the processed list
+                    processedGroups.add(group);
+                    // add the usage clause
+                    buff.append(pfx).append(asSyntaxOptions(group));
+                    pfx = " ";
+                }
+                // otherwise the option was displayed in the group previously so ignore it.
+            }
+            // if the Option is not part of an OptionGroup
+            else {
+                buff.append(pfx).append(optionFormatBuilder.build(option).asSyntaxOption());
+                pfx = " ";
+            }
+        }
+        return buff.toString();
+    }
+
+    /**
+     * Return the string representation of the options as used in the syntax display.
+     * @param group The OptionGroup to create the string representation for.
+     * @return the string representation of the options as used in the syntax display.
+     */
+    public String asSyntaxOptions(final OptionGroup group) {
```

Comment:
Java tends to use the "to" prefix for conversions; I've seen "as" used for type change methods. For example `Object.toString()` to convert an Object to a String (obviously) and `Class.asSubclass(...)` to type cast one class to another. So I think all of our methods to convert something to a String should be "to" methods. 

WDYT?

Reply:
I don't this is line with most of the existing JDK naming and I do not think we should stray too far from those.

For example, the API is `String.toLowerCase()` and `toUpperCase()`, _not_ `asLowerCase()`. See also my previous `toString()` example which is _all_ about representation. See also `URL.toASCIIString()`, `Duration.to*()`, and so on.

The examples in the JDK for "as" methods usually return a different view on an underlying object, which is not the case here, since we are returning a new immutable Object, a String. For example, ALL implementations of `java.nio.CharBuffer.asReadOnlyBuffer()`, `java.nio.ByteBuffer.asDoubleBuffer()` (`java.nio.ByteBuffer.as*Buffer()`) are views where "The content of the new buffer will be that of this buffer. Changes to this buffer's content will be visible in the new buffer". 

There probably "as" methods in the JDK that have different semantic but there are so many "to" methods that this should be familiar territory for all users.